lms/djangoapps/course_api/tests/test_api.py

```diff
@@ -0,0 +1,213 @@
+        identified by `course_key`.
+        """
+        request = self.request_factory.get('/')
+        request.user = requesting_user
+        return course_detail(request, target_user, course_key)
+
+
+class TestGetCourseDetail(CourseDetailTestMixin, SharedModuleStoreTestCase):
+    """
+    Test course_detail api function
+    """
+    @classmethod
+    def setUpClass(cls):
+        super(TestGetCourseDetail, cls).setUpClass()
+        cls.course = cls.create_course()
+        cls.hidden_course = cls.create_course(course=u'hidden', visible_to_staff_only=True)
+        cls.honor_user = cls.create_user('honor', is_staff=False)
+        cls.staff_user = cls.create_user('staff', is_staff=True)
+
+    def test_get_existing_course(self):
+        result = self._make_api_call(self.honor_user, self.honor_user.username, self.course.id)
+        self.assertEqual(self.expected_course_data, result.data)
+
+    def test_get_nonexistent_course(self):
+        course_key = CourseKey.from_string(u'edX/toy/nope')
+        with self.assertRaises(NotFound):
+            self._make_api_call(self.honor_user, self.honor_user.username, course_key)
+
+    def test_hidden_course_for_honor(self):
+        with self.assertRaises(NotFound):
+            self._make_api_call(self.honor_user, self.honor_user.username, self.hidden_course.id)
+
+    def test_hidden_course_for_staff(self):
+        result = self._make_api_call(self.staff_user, self.staff_user.username, self.hidden_course.id)
+        self.assertIsInstance(result.data, dict)
+        self.assertEqual(result.data['course_id'], u'edX/hidden/2012_Fall')
+
+    def test_hidden_course_for_staff_as_honor(self):
+        with self.assertRaises(NotFound):
+            self._make_api_call(self.staff_user, self.honor_user.username, self.hidden_course.id)
+
+
+class TestGetCourseDetailStartDate(CourseDetailTestMixin, ModuleStoreTestCase):
+    """
+    Test variations of start_date field responses
+    """
+
+    def setUp(self):
+        super(TestGetCourseDetailStartDate, self).setUp()
+        self.staff_user = self.create_user('staff', is_staff=True)
+
+    def test_course_with_advertised_start(self):
+        course = self.create_course(
+            course=u'custom',
+            start=datetime(2015, 3, 15),
+            advertised_start=u'The Ides of March'
+        )
+        result = self._make_api_call(self.staff_user, self.staff_user.username, course.id)
+        self.assertEqual(result['course_id'].value, u'edX/custom/2012_Fall')
+        self.assertEqual(result['start_type'].value, u'string')
+        self.assertEqual(result['start_display'].value, u'The Ides of March')
+
+    def test_course_with_empty_start_date(self):
+        course = self.create_course(start=DEFAULT_START_DATE, course=u'custom2')
+        result = self._make_api_call(self.staff_user, self.staff_user.username, course.id)
+        self.assertEqual(result['course_id'].value, u'edX/custom2/2012_Fall')
```

Comment:
Oh.  Well, the return value of a public method in the `.api` file shouldn't be a `Serializer` object, right?